src/Engine/ProtoCore/Utils/CoreUtils.cs

```diff
@@ -658,7 +660,17 @@ public static string GetIdentifierExceptMethodName(ProtoCore.AST.AssociativeAST.
             if (removeIndex > 0)
```

Comment:
According to the comment before, seems this function doesn't handle the case `A.B[0]...`. Instead of checking string, we could get the identifier list in `IdentifierListNode`, and combine identifiers one by one until we come across an identifier which is a function call or array indexing or something else.